src/CommitmentsV2/SFA.DAS.CommitmentsV2/Application/Queries/GetApprenticeships/GetApprenticeshipsHandler.cs

```diff
@@ -26,24 +28,253 @@ public class GetApprenticeshipsHandler : IRequestHandler<GetApprenticeshipsReque
 
-            var matched = await _dbContext.Apprenticeships
+            ApprenticeshipSearchResult searchResult;
+
+            if (!string.IsNullOrEmpty(request.SortField) && request.SortField != nameof(Apprenticeship.DataLockStatus) && request.ReverseSort)
+            {
+                searchResult = await ApprenticeshipsReverseOrderedByField(cancellationToken, request.ProviderId, request.SortField, request.PageNumber, request.PageItemCount);
+            }
+            else if (!string.IsNullOrEmpty(request.SortField) && request.SortField != nameof(Apprenticeship.DataLockStatus))
+            {
+                searchResult = await ApprenticeshipsOrderedByField(cancellationToken, request.ProviderId, request.SortField, request.PageNumber, request.PageItemCount);
+            }
+            else if (string.IsNullOrEmpty(request.SortField) && request.ReverseSort)
+            {
+                searchResult = await ApprenticeshipsByReverseDefaultOrder(cancellationToken, request.ProviderId, request.PageNumber, request.PageItemCount);
+            }
+            else
+            {
+                searchResult = await ApprenticeshipsByDefaultOrder(cancellationToken, request.ProviderId, request.PageNumber, request.PageItemCount);
+            }
+
+            foreach (var apprenticeship in searchResult.Apprenticeships)
+            {
+                var details = await _mapper.Map(apprenticeship);
+                matchedApprenticeshipDetails.Add(details);
+            }
+
+            return new GetApprenticeshipsResponse
+            {
+                Apprenticeships = matchedApprenticeshipDetails,
+                TotalApprenticeshipsFound = searchResult.TotalApprenticeshipsFound,
+                TotalApprenticeshipsWithAlertsFound = searchResult.TotalApprenticeshipsWithAlertsFound
+            };
+        }
+
+        private async Task<ApprenticeshipSearchResult> ApprenticeshipsByDefaultOrder(CancellationToken cancellationToken, long? providerId, int pageNumber, int pageItemCount)
+        {
+            var apprenticeships = await _dbContext
+                .Apprenticeships
+                .Where(apprenticeship => apprenticeship.Cohort.ProviderId == providerId && apprenticeship.PendingUpdateOriginator != null)
+                .OrderBy(x => x.FirstName)
+                .ThenBy(x => x.LastName)
+                .ThenBy(x => x.Uln)
+                .ThenBy(x => x.Cohort.LegalEntityName)
+                .ThenBy(x => x.CourseName)
+                .ThenByDescending(x => x.StartDate)
+                .Include(apprenticeship => apprenticeship.Cohort)
+                .Include(apprenticeship => apprenticeship.DataLockStatus)
+                .ToListAsync(cancellationToken);
+            
+            var apprenticeshipsWithoutAlerts = await _dbContext
+                .Apprenticeships
+                .Where(apprenticeship => apprenticeship.Cohort.ProviderId == providerId && apprenticeship.PendingUpdateOriginator == null)
```

Comment:
Although using `PendingUpdateOriginator` will work, unfortunately it's earmarked for removal. Could you join to the `ApprenticeshipUpdate` collection instead?

Reply:
Spoke to Dan as originally we thought this was how we could determine that an alert was present. He's working on updating this code now to reflect the changes required